wwwapp/views.py

```diff
@@ -793,3 +807,32 @@ def upload_file(request, type, name):
             destination.write(chunk)
 
     return JsonResponse({'location': urljoin(urljoin(settings.MEDIA_URL, target_dir), name)})
+
+
+@login_required()
+def cloud_access_view(request):
+    context = get_context(request)
+    userprofile = request.user.userprofile
+    is_lecturer = userprofile.is_lecturer_in(settings.CURRENT_YEAR)
+    is_participant = userprofile.is_participant_in(settings.CURRENT_YEAR)
+    if not (is_lecturer or is_participant):
+        return HttpResponseForbidden()
+
+    if userprofile.owncloud_user == "":
+        userprofile.owncloud_user = \
+            unicodedata.normalize("NFKD", request.user.first_name+request.user.last_name)\
+                .encode('ascii','ignore').decode('ascii')[:32]
+        userprofile.owncloud_password = generate_nice_pass()
+        c = owncloud.Client(settings.OWNCLOUD_HOST)
+        c.login(settings.OWNCLOUD_USER, settings.OWNCLOUD_PASS)
+        c.create_user(userprofile.owncloud_user, userprofile.owncloud_password)
+        if is_lecturer:
+            c.add_user_to_group(userprofile.owncloud_user, "WWW16Lecturers")
```

Comment:
This should probably be configurable as well, but we can address this later (add an issue sowe wo't forget)

Reply:
Who cares? This code is only for this year.